### ISSUE TITLE
Dynamic import support

### ISSUE THREAD

**Issue by guybedford** [systemjs/builder#813]
Support dynamic import syntax through the Babel compiler.

This should probably disable Rollup optimizations for that specific module as Rollup doesn't support dynamic imports.

Then this should also be added to the plugin ecosystem as well.

**Comment by guybedford** [comment]
First attempt in https://github.com/systemjs/builder/commit/772382744a4eb6e51b47c0865c21bcc2cafedcc6.

**Comment by MrJoe** [comment]
What level of support for dynamic import is currently supported? (If any)

For example I noticed the following somewhat works

```
import * as x from "somemodule";

// whatever some code here
// followed by a dynamic import later in the file:
import("othermodule").then(x => { /* dosomething */ });
```

the othermodule is transpiled to `_context.import("othermodule").then ...` however normalization of the module does not work. So it didn't became `_content.import("npm:othermodule:0.0.0/lib.js").then...`

**Comment by IceCreamYou** [comment]
Is any support for dynamic imports via `System.import()` planned, or only ES6 `import()` syntax?

**Comment by cancio** [comment]
Is there any support for dynamic imports via System.import() currently? In other words, when using the builder, in the final bundle am I supposed to see the modules that are dynamically imported:

**Comment by amenadiel** [comment]
Nope. Modules imported with System.import  are not considered as dependencies of the script calling them.

Which is not necessarily bad specially if you import a huge library inside a función call.

**Comment by javadbat** [comment]
I'm trying to build react router that works with systemjs and webpack together and i need dynamic import to do so. does anyone know any solution for that? 
my router currently work with systemjs "System.import" to publishing it on npm i need to make it work with webpack too
